coq/base_file.py

```diff
@@ -403,13 +413,10 @@ def traverse_expr(expr):
                 "Hypotheses",
             ]:
                 if text.startswith(keyword):
+                    self._last_term = Term(
```

Comment:
Add a comment here explaining why we are doing this and why does it work as a hotfix. Otherwise in the future we will be confused (we already have experience in that situation).